caffe2/core/dispatch/DispatchKey.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "caffe2/core/dispatch/DeviceId.h"
+#include "caffe2/core/dispatch/LayoutId.h"
+#include "caffe2/core/typeid.h"
+
+#include <vector>
+#include <functional>
+#include "caffe2/utils/Array.h"
+
+namespace c10 {
+
+namespace details {
+struct TensorParameterDispatchKey final {
+  DeviceId deviceId;
+  LayoutId layoutId;
```

Comment:
so in the new design we discussed, LayoutId would go away, right?

Reply:
yes, there's going to be a PR for that soon